src/app/containers/MostRead/Canonical/index.jsx

```diff
@@ -42,90 +34,42 @@ const MarginWrapper = styled.div`
-    margin: 0 ${GEL_MARGIN_ABOVE_400PX} ${GEL_SPACING_QUAD};
-  }
-  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MAX}) {
-    margin: 0 ${GEL_MARGIN_ABOVE_400PX} ${GEL_SPACING_QUIN};
-  }
-  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
-    width: 100%; /* Needed for IE11 */
-    margin: 0 auto ${GEL_SPACING_TRPL};
-    padding: 0 ${GEL_SPACING_DBL};
-    max-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN};
-  }
-`;
-
 const CanonicalMostRead = ({
   endpoint,
   columnLayout,
-  constrainMaxWidth,
-  isOnFrontPage,
+  initialData,
+  wrapper: Wrapper,
 }) => {
-  const [items, setItems] = useState([]);
   const {
     service,
     script,
     dir,
     datetimeLocale,
     timezone,
-    mostRead: { header, lastUpdated, numberOfItems },
+    mostRead: { lastUpdated, numberOfItems },
   } = useContext(ServiceContext);
 
-  useEffect(() => {
-    const handleResponse = async response => {
-      const mostReadData = await response.json();
+  const filteredData = processMostRead({ data: initialData, numberOfItems });
 
-      // The ARES test endpoint for most read renders fixture data, so the data is stale
-      const isTest = process.env.SIMORGH_APP_ENV === 'test';
+  const [items, setItems] = useState(filteredData);
 
-      // Do not show most read if lastRecordUpdated is greater than 35min as this means PopAPI has failed twice
-      // in succession. This suggests ATI may be having issues, hence risk of stale data.
-      if (isTest || mostReadRecordIsFresh(mostReadData.lastRecordTimeStamp)) {
-        const mostReadItems = mostReadData.records
-          .slice(0, numberOfItems)
-          .map(({ id, promo: { headlines, locators, timestamp } }) => ({
-            id,
-            title: headlines.shortHeadline,
-            href: locators.assetUri,
-            timestamp: shouldRenderLastUpdated(timestamp) && (
-              <LastUpdated
-                prefix={lastUpdated}
-                script={script}
-                service={service}
-                timestamp={timestamp}
-                locale={datetimeLocale}
-                timezone={timezone}
-              />
-            ),
-          }));
-        setItems(mostReadItems);
-      }
-    };
-    const fetchMostReadData = pathname =>
-      fetch(pathname, { mode: 'no-cors' })
-        .then(handleResponse)
-        .catch(e => logger.error(`HTTP Error: "${e}"`));
-    fetchMostReadData(endpoint);
+  useEffect(() => {
+    if (!items) {
+      const handleResponse = async response => {
+        if (!response.ok) {
+          throw Error(response.statusText);
+        }
+        const mostReadData = await response.json();
+        setItems(processMostRead({ data: mostReadData, numberOfItems }));
+      };
+
+      const fetchMostReadData = pathname =>
+        fetch(pathname, { mode: 'no-cors' })
+          .then(handleResponse)
+          .catch(e => logger.error(`HTTP Error: "${e}"`));
```

Comment:
Shall we log the error here as well, as we've done with Radio Schedules? https://github.com/bbc/simorgh/pull/6142/files#diff-91ab1611f147d3d33e441ceb72157ed3R143

Reply:
yeah definitely, I wanted to hold out doing it in this PR as I wanted to make it the same as the radio schedules container so I'll wait till that is merged.

We have this issue: https://github.com/bbc/simorgh/issues/6219 to consolidate the two containers after the work has been merged.